src/middlewares/AuthMiddleware.js

```diff
@@ -0,0 +1,24 @@
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+
+exports.protectedUser = async (req, res, next) => {
+  if (
+    !req.headers.authorization ||
+    !req.headers.authorization.startsWith("Bearer")
+  ) {
+    return res.status(401).json({
+      status: false,
+      data: [],
+      message: "Unauthorized access",
+    });
+  }
+  const token = req.headers.authorization.split(" ")[1];
+
+  try {
+    const decoded = jwt.verify(token, process.env.JWT_SECRET_KEY);
+    req.user = decoded;
```

Comment:
req.user = {
   email: decoded.userEmail,
  id: decoded.userId
}

do this instead

Reply:
you can do the validation of if req.user.email exists in db here, if it does, proceed, if it doesn't throw an unauthorized error